syncmymoodle/__main__.py

```diff
@@ -197,7 +197,7 @@ def get_session_key(soup):
         resp = self.session.get(
             "https://moodle.rwth-aachen.de/auth/shibboleth/index.php"
         )
-        if resp.url == "https://moodle.rwth-aachen.de/my/":
+        if "https://moodle.rwth-aachen.de/" in resp.url:
```

Comment:
```suggestion
        if resp.url.startswith("https://moodle.rwth-aachen.de/my/"):
```

Reply:
As you can see the login page lives under /auth/, it is very likely that errors during authentication are also presented under that directory (e.g. Moodle gets a session from Shibboleth which is missing some af the attributes it expects)